src/lib/writer.h

```diff
@@ -0,0 +1,75 @@
+/*
+ * This file is part of clr-boot-manager.
+ *
+ * Copyright © 2017 Intel Corporation
+ *
+ * clr-boot-manager is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public License as
+ * published by the Free Software Foundation; either version 2.1
+ * of the License, or (at your option) any later version.
+ */
+
+#pragma once
+
+#include "nica/util.h"
+
+#define _GNU_SOURCE
+
+typedef struct CbmWriter {
+        FILE *memstream;
+        char *buffer;
+        size_t buffer_n;
+        int error;
+} CbmWriter;
+
+#define CBM_WRITER_INIT &(CbmWriter){ 0 };
```

Comment:
Just as long as everybody knows that's not really a pointer and if the stack it was allocated on goes away, you can't use it anymore.